File: 60permutationSequence/permutationSequence.py
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        factorial = math.factorial(n)
        result, k = [], k - 1
        *elements, = range(1, n + 1)
        while elements:
            factorial //= len(elements)
            _, k = divmod(k, factorial)
            result += elements.pop(_),
        return ''.join(map(str, result))

File: 60permutationSequence/permutationSequence.js
/**
 * @param {number} n
 * @param {number} k
 * @return {string}
 */
var getPermutation = function(n, k) {
    let factorial = 1
    for (let _ = 2; !Object.is(_, n + 1); ++_) factorial *= _
    const result = []
    --k
    let elements = [...Array(n + 1).keys()].slice(1)
    while (elements.length)
    {
        factorial /= elements.length;
        [_, k] = [Math.trunc(k / factorial), k % factorial]
        result.push(elements.splice(_, 1))
    }
    return result.join('')
};
